text-feature-translation/src/main/java/net/kyori/adventure/text/feature/translation/TranslationRegistryImpl.java

```diff
@@ -125,12 +137,15 @@
 
     // no arguments makes this render very simple
     if(args.isEmpty()) {
-      return builder.content(format.format(null, new StringBuffer(), null).toString()).build();
+      return builder.content(format).build();
     }
 
+    // TODO: MessageFormat has unnecessary overhead over String, rewrite this code to use String
+    final MessageFormat messageFormat = new MessageFormat(format);
```

Comment:
Why not have the translator return a MessageFormat? It should be possible to have a ResourceBundle instance that caches MessageFormat instances (probably within a ThreadLocal since the formats are not thread-safe).